lib/device-sockets/ios/socket-request-executor.ts

```diff
@@ -31,37 +27,38 @@ export class IOSSocketRequestExecutor implements IiOSSocketRequestExecutor {
 				this.$errors.failWithoutHelp("A client is already connected.");
 				break;
 			case this.$iOSNotification.attachAvailable:
-				await this.executeAttachAvailable(npc, timeout);
+				await this.executeAttachAvailable(device.deviceInfo.identifier, timeout);
 				break;
 			case this.$iOSNotification.readyForAttach:
 				break;
 		}
 	}
 
-	public async executeLaunchRequest(device: Mobile.IiOSDevice, timeout: number, readyForAttachTimeout: number, shouldBreak?: boolean): Promise<void> {
-		let npc = new iOSProxyServices.NotificationProxyClient(device, this.$injector);
-
+	public async executeLaunchRequest(deviceIdentifier: string, timeout: number, readyForAttachTimeout: number, shouldBreak?: boolean): Promise<void> {
 		try {
-			await this.$iOSNotificationService.awaitNotification(npc, this.$iOSNotification.appLaunching, timeout);
-			process.nextTick(() => {
-				if (shouldBreak) {
-					npc.postNotificationAndAttachForData(this.$iOSNotification.waitForDebug);
-				}
-
-				npc.postNotificationAndAttachForData(this.$iOSNotification.attachRequest);
-			});
+			// We should create these promises here beecause we need to send the ObserveNotification on the device
+			// before we send the PostNotification.
+			const promisesToWait = [this.$iOSNotificationService.awaitNotification(deviceIdentifier, this.$iOSNotification.appLaunching, timeout),
+			this.$iOSNotificationService.awaitNotification(deviceIdentifier, this.$iOSNotification.readyForAttach, readyForAttachTimeout)];
+			if (shouldBreak) {
+				await this.$iOSNotificationService.postNotificationAndAttachForData(deviceIdentifier, this.$iOSNotification.waitForDebug);
+			}
 
-			await this.$iOSNotificationService.awaitNotification(npc, this.$iOSNotification.readyForAttach, readyForAttachTimeout);
+			await this.$iOSNotificationService.postNotificationAndAttachForData(deviceIdentifier, this.$iOSNotification.attachRequest);
+			await Promise.all(promisesToWait);
```

Comment:
hm, is `Promise.all` the correct method here. The order before was:
- appLaunching
- readyForAttach
- waitForDebug (optional)
- attachRequest

Now, with current code, it is:
- appLaunching, readyForAttach (simultaneously)
- waitForDebug
- attachRequest

Or even in some cases:
- waitForDebug
- attachRequest
- appLaunching, readyForAttach (simultaneously)

Can you elaborate why is the change of the order?